components/style/values/specified/length.rs

```diff
@@ -758,8 +875,8 @@ impl CalcLengthOrPercentage {
             match value {
                 SimplifiedValueNode::Percentage(p) =>
                     percentage = Some(percentage.unwrap_or(0.) + p),
-                SimplifiedValueNode::Length(NoCalcLength::Absolute(Au(au))) =>
-                    absolute = Some(absolute.unwrap_or(0) + au),
+                SimplifiedValueNode::Length(NoCalcLength::Absolute(length)) =>
+                    absolute = Some(absolute.unwrap_or(0) + Au::from(length).0),
```

Comment:
Let's just convert everything to the same unit in `calc()`, (pixels seems adequate), and making `absolute` get a floating point value. You can then do:

```rust
absolute = Some(absoulte.unwrap_or(0.) + Au::from(length).to_f32_px())
```